cou/utils/juju_utils.py

```diff
@@ -180,9 +180,9 @@ async def _get_application(self, name: str) -> Application:
 
         :param name: Name of application
         :type name: str
+        :raises ApplicationNotFound: When cannot find Application in the model.
```

Comment:
nit: This sounds a bit weird. Would it be better like `When Application is not found in the model`